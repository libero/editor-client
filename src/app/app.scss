.app-container {
  max-width: 100%;
  margin: 0 auto;
  height: 100%;
}
