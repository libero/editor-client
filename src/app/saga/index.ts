import { fork, all, AllEffect, ForkEffect, put } from 'redux-saga/effects';
import manuscriptSaga from './manuscript.saga';
import formattersSaga from './formatters.saga';
import manuscriptEditorSaga from './manuscript-editor.saga';
import syncChangesSaga from './sync.saga';
import tocSaga from './toc.saga';
import { routerSaga } from './router.saga';
import { LocalStorageApi } from '../api/local-storage.api';
import * as manuscriptEditorAction from '../actions/manuscript-editor.actions';

// eslint-disable-next-line
function* initialSaga<T>() {
  const taskId = LocalStorageApi.get(LocalStorageApi.EXPORT_TASK_KEY);
  if (typeof taskId === 'string') {
    yield put(manuscriptEditorAction.setActiveExportPdfTask(taskId));
  }
}

export function* rootSaga(): IterableIterator<AllEffect<ForkEffect>> {
  yield all([
    fork(initialSaga),
    fork(routerSaga),
    fork(syncChangesSaga),
    fork(formattersSaga),
    fork(manuscriptSaga),
    fork(manuscriptEditorSaga),
    fork(tocSaga)
  ]);
}
