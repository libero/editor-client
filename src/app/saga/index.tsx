import { fork, all } from 'redux-saga/effects';
import manuscriptSaga from './manuscript.saga';
import { routerSaga } from '../saga/router.saga';

// eslint-disable-next-line require-yield
function* initialSaga() {
  console.log('Initialising');
}

export function* rootSaga() {
  yield all([fork(initialSaga), fork(routerSaga), fork(manuscriptSaga)]);
}
