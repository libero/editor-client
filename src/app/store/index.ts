import { combineReducers, createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension';
import { rootSaga } from '../saga';
import { manuscriptReducer } from '../reducers/manuscript.reducer';
import { LoadableState, ManuscriptHistory } from '../utils/state.utils';
import { manuscriptEditorReducer } from '../reducers/manuscript-editor.reducer';

const sagaMiddleware = createSagaMiddleware();

export type ManuscriptHistoryState = LoadableState<ManuscriptHistory>;

export interface ManuscriptEditorState {
  focusedManuscriptPath: string | undefined;
}

export interface ApplicationState {
  manuscript: ManuscriptHistoryState;
  manuscriptEditor: ManuscriptEditorState;
}

export const store = createStore(
  combineReducers({
    manuscript: manuscriptReducer,
    manuscriptEditor: manuscriptEditorReducer
  }),
  composeWithDevTools(applyMiddleware(sagaMiddleware))
);

sagaMiddleware.run(rootSaga);
