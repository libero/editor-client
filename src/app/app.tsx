import React from 'react';

import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from './store';
import './app.scss';
import { ManuscriptContainer } from './containers/manuscript';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <Router>
        <div className="app-container">
          <Switch>
            <Route path="/">
              <ManuscriptContainer />
            </Route>
          </Switch>
        </div>
      </Router>
    </Provider>
  );
};
