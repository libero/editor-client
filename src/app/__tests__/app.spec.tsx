import React from 'react';
import { create } from 'react-test-renderer';
import { App } from '../app';

jest.mock('react-redux', () => {
  return {
    Provider: ({ children }) => <>{children}</>
  };
});

jest.mock('react-router-dom', () => {
  return {
    BrowserRouter: ({ children }) => <div data-cmp="BrowserRouter">{children}</div>,
    Switch: ({ children }) => <div data-cmp="Switch">{children}</div>,
    Route: ({ children }) => <div data-cmp="Route">{children}</div>
  };
});

jest.mock('../containers/manuscript', () => {
  return {
    ManuscriptContainer: () => <div data-cmp="manuscript"></div>
  };
});

describe('App container', () => {
  it('renders', () => {
    const component = create(<App />);
    expect(component).toMatchSnapshot();
  });
});
