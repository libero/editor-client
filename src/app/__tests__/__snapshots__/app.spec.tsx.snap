// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container renders 1`] = `
<div
  data-cmp="ConnectedRouter"
>
  <main
    className="app-container"
    role="application"
  >
    <div
      data-cmp="Switch"
    >
      <div
        data-cmp="Route"
      >
        <div
          data-cmp="manuscript"
        />
      </div>
    </div>
  </main>
</div>
`;
